pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/DefaultCommitterRealtimeIntegrationTest.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.pinot.common.metadata.segment.RealtimeSegmentZKMetadata;
+import org.apache.pinot.common.metrics.ServerMeter;
+import org.apache.pinot.common.metrics.ServerMetrics;
+import org.apache.pinot.common.protocols.SegmentCompletionProtocol;
+import org.apache.pinot.common.utils.TarGzCompressionUtils;
+import org.apache.pinot.controller.helix.core.realtime.SegmentCompletionManager;
+import org.apache.pinot.core.data.manager.realtime.LLRealtimeSegmentDataManager;
+import org.apache.pinot.core.data.manager.realtime.SegmentCommitter;
+import org.apache.pinot.core.data.manager.realtime.SegmentCommitterFactory;
+import org.apache.pinot.core.indexsegment.generator.SegmentGeneratorConfig;
+import org.apache.pinot.core.segment.creator.SegmentIndexCreationDriver;
+import org.apache.pinot.core.segment.creator.impl.SegmentIndexCreationDriverImpl;
+import org.apache.pinot.core.segment.index.loader.IndexLoadingConfig;
+import org.apache.pinot.server.realtime.ControllerLeaderLocator;
+import org.apache.pinot.server.realtime.ServerSegmentCompletionProtocolHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Integration test that extends RealtimeClusterIntegrationTest but uses low-level Kafka consumer.
+ * TODO: Add separate module-level tests and remove the randomness of this test
+ */
+public class DefaultCommitterRealtimeIntegrationTest extends RealtimeClusterIntegrationTest {
+  private static final String CONSUMER_DIRECTORY = "/tmp/consumer-test";
+  private static final File INDEX_DIR = new File(FileUtils.getTempDirectory(), "SegmentCommitterTest");
+  private static final String AVRO_DATA = "data" + File.separator + "test_data-sv.avro";
+  private final boolean _enableSplitCommit = false;
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultCommitterRealtimeIntegrationTest.class);
+  private final long END_OFFSET = 500L;
+  private File TEMP_FILE;
+  private static final String SEGMENT_NAME = "testTable_126164076_167572854";
+  private File untarredSegmentFile = new File(INDEX_DIR, SEGMENT_NAME);
+  private File _realtimeSegmentUntarred;
+
+  @BeforeClass
+  @Override
+  public void setUp()
+      throws Exception {
+    TEMP_FILE = File.createTempFile("SegmentCommitterTest", null);
+    // Remove the consumer directory
+    File consumerDirectory = new File(CONSUMER_DIRECTORY);
+    if (consumerDirectory.exists()) {
+      FileUtils.deleteDirectory(consumerDirectory);
+    }
+
+    startZk();
+    startController();
+    addFakeBrokerInstancesToAutoJoinHelixCluster(1, true);
+    addFakeServerInstancesToAutoJoinHelixCluster(1, true);
+
+    ControllerLeaderLocator.create(_helixManager);
+
+    // Start Kafka
+    startKafka();
+    List<File> avroFiles = unpackAvroData(_tempDir);
```

Comment:
checkin 2 or 3  segments with 1 or 2 rows in them that is already built. That will save run time. Maybe we already have some checked in segments, check first.